cypress/integration/backoffice/sor-contracts.spec.js

```diff
@@ -0,0 +1,186 @@
+/// <reference types="cypress" />
+
+import 'cypress-audit/commands'
+
+describe('SOR-Contracts - when user unauthorized', () => {
+  it("Shows access denied when user doesn't have correct permissions", () => {
+    cy.loginWithOperativeRole()
+    cy.visit('/backoffice/sor-contracts')
+    cy.contains('Access denied')
+  })
+})
+
+describe('SOR-Contracts - When Copy selected', () => {
+  beforeEach(() => {
+    cy.intercept(
+      {
+        method: 'GET',
+        path: '/api/contractors?getAllContractors=true',
+      },
+      {
+        fixture: 'contractors/contractors.json',
+      }
+    ).as('contractorsRequest')
+
+    cy.loginWithDataAdminRole()
+    cy.visit('/backoffice/sor-contracts')
+
+    cy.wait('@contractorsRequest')
+  })
+
+  it('Shows error messages when form fields invalid', () => {
+    cy.get("[data-test='submit-button']").click()
+
+    cy.get('.govuk-error-message.lbh-error-message').contains(
+      'You must enter a source property reference'
+    )
+    cy.get('.govuk-error-message.lbh-error-message').contains(
+      'You must enter a destination property reference'
+    )
+  })
+
+  it('shows error messages when property references invalid', () => {
+    const invalidPropertyReference = 'abc1234'
+    cy.get('[data-test="sourcePropertyReference"]').type(
+      invalidPropertyReference
+    )
+    cy.get('[data-test="destinationPropertyReference"]').type(
+      invalidPropertyReference
+    )
+
+    cy.get("[data-test='submit-button']").click()
+
+    cy.get('.govuk-error-message.lbh-error-message')
+      .filter(':contains("PropertyReference is invalid")')
+      .should('have.length', 2)
+  })
+
+  it('shows an error message when both property references match', () => {
+    const validPropertyReference = '12345678'
+    cy.get('[data-test="sourcePropertyReference"]').type(validPropertyReference)
+    cy.get('[data-test="destinationPropertyReference"]').type(
+      validPropertyReference
+    )
+
+    cy.get("[data-test='submit-button']").click()
+
+    cy.get('.govuk-error-message.lbh-error-message').contains(
+      'The destination property reference cannot match source property reference'
+    )
+  })
+
+  it('sends request to /sor-contracts', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        path: '/api/backOffice/sor-contracts',
+      },
+      {
+        body: '',
+      }
+    ).as('sorContractRequest')
+
+    const propertyReference1 = '12345678'
+    const propertyReference2 = '87654321'
+    cy.get('[data-test="sourcePropertyReference"]').type(propertyReference1)
+    cy.get('[data-test="destinationPropertyReference"]').type(
+      propertyReference2
+    )
+    cy.get("[data-test='submit-button']").click()
+
+    cy.wait('@sorContractRequest')
+
+    cy.contains('SOR Contract Modification')
+    cy.contains('SOR contracts modified successfully!')
```

Comment:
In my understanding the cy.contains itself is not an assertion, but it's used to get HTML elements. If you want to assert that the text "SOR Contract Modification" is shown I'd suggest adding a "should" after it:

example: cy.contains('SOR Contract Modification').should('be.visible')

or you could do something like 

example: cy.get('[data-test="blabla"]').should('have.text', 'SOR Contract Modification')

I'd apply this to every cy.contains in this file that's been used as an assertion